File: models/Payment.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const PaymentSchema = new Schema({
  ticket:   { type: Schema.Types.ObjectId, ref: 'Ticket', required: true, index: true },
  user:     { type: Schema.Types.ObjectId, ref: 'User', required: true, index: true },

  method:   { type: String, enum: ['cash','momo','zalopay','vnpay','card','unknown'], default: 'unknown' },
  amount:   { type: Number, required: true, min: 0 },

  status:   { type: String, enum: ['pending','succeeded','failed','refunded'], default: 'pending', index: true },
  provider_txn_id:   { type: String, default: '' },
  provider_message:  { type: String, default: '' },
  meta:     { type: Schema.Types.Mixed, default: {} }
}, { timestamps: true });

module.exports = mongoose.model('Payment', PaymentSchema);

File: app/src/main/java/com/example/datn_md_13/Model/Payment.java
package com.example.datn_md_13.Model;

public class Payment {
    public String _id;
    public String ticket;
    public String user;
    public String method;             // cash | momo | zalopay | vnpay | card | unknown
    public double amount;
    public String status;             // pending | succeeded | failed | refunded
    public String provider_txn_id;    // mã giao dịch của VNPAY
    public String provider_message;   // thông báo từ provider
    public Object meta;               // object linh hoạt
    public String createdAt;
    public String updatedAt;
}
